### PR TITLE
add quickcheck tests and fix off-by-one bug introduced in #31

### PR DESCRIPTION
**Pull request by alindima**
we have fuzzers but they're not being run in the CI (and I couldn't even get them to run locally, I have to check why).
quickcheck will help spot bugs in the CI

Fixes a bug introduced in https://github.com/paritytech/reed-solomon-novelpoly/pull/31, quickcheck aided in identifying it so it will act as a regression test also

### CODE REVIEW

File: reed-solomon-novelpoly/src/field/inc_encode.rs [hunk 1]
```diff
@@ -190,7 +190,9 @@ pub fn encode_sub_plain(bytes: &[u8], n: usize, k: usize) -> Result<Vec<Additive
 	// so we get a buffer of size `N` in `GF` symbols
 	let mut elm_data = vec![Additive(0); n];
 
-	for i in 0..(bytes_len / 2) {
+	let tmp = if bytes_len % 2 == 0 { bytes_len / 2 } else { (bytes_len + 1) / 2 };
```
Comment (ordian) [inline reed-solomon-novelpoly/src/field/inc_encode.rs]: can it be simplified to
```suggestion
	let tmp = (bytes_len + 1) / 2;
```

File: reed-solomon-novelpoly/src/field/inc_encode.rs [hunk 2]
```diff
@@ -241,7 +243,9 @@ pub fn encode_sub_faster8(bytes: &[u8], n: usize, k: usize) -> Result<Vec<Additi
...
```
Comment (ordian) [inline reed-solomon-novelpoly/src/field/inc_encode.rs]: and here as well
```suggestion
	let tmp = (bytes_len + 1) / 2;
```

File: reed-solomon-novelpoly/src/novel_poly_basis/tests.rs [hunk 3]
```diff
@@ -463,3 +464,32 @@ fn shard_len_is_reasonable() {
...
 	assert_eq!(rs.shard_len(19), 6);
 }
+
+#[derive(Clone, Debug)]
+struct ArbitraryData(Vec<u8>);
+
+impl Arbitrary for ArbitraryData {
+	fn arbitrary(g: &mut Gen) -> Self {
+		// Limit the len to 1 mib, otherwise the test will take forever
+		let len = u32::arbitrary(g).saturating_add(2) % (1024 * 1024);
```
Comment (ordian) [inline reed-solomon-novelpoly/src/novel_poly_basis/tests.rs]: what is this saturating add for? if we want len at least 2, it should be added after modulus operation or simply .max(2)

Comment (alindima) [reply reed-solomon-novelpoly/src/novel_poly_basis/tests.rs]: yes, to not get a zeroed length.
doesn't that achieve exactly the same thing as my code?

Comment (ordian) [reply reed-solomon-novelpoly/src/novel_poly_basis/tests.rs]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=aa3f3dd2f1cd8d82eea0b80c65b3a3bb

Comment (alindima) [reply reed-solomon-novelpoly/src/novel_poly_basis/tests.rs]: didn't think of this 👍🏻 thanks
